src/gui/systray.cpp

```diff
@@ -133,123 +126,225 @@ void Systray::setToolTip(const QString &tip)
     QSystemTrayIcon::setToolTip(tr("%1: %2").arg(Theme::instance()->appNameGUI(), tip));
 }
 
-int Systray::calcTrayWindowX()
+bool Systray::syncIsPaused()
 {
-#ifdef Q_OS_OSX
-    // macOS handles DPI awareness differently
-    // and menu bar is always at the top, icons starting from the right
+    return _syncIsPaused;
+}
 
-    QPoint topLeft = this->geometry().topLeft();
-    QPoint topRight = this->geometry().topRight();
-    int trayIconTopCenterX = (topRight - ((topRight - topLeft) * 0.5)).x();
-    return trayIconTopCenterX - (400 * 0.5);
-#else
-    QScreen* trayScreen = nullptr;
-    if (QGuiApplication::screens().count() > 1) {
-        trayScreen = QGuiApplication::screens().at(screenIndex());
+void Systray::pauseResumeSync()
+{
+    if (_syncIsPaused) {
+        _syncIsPaused = false;
+        emit resumeSync();
     } else {
-        trayScreen = QGuiApplication::primaryScreen();
+        _syncIsPaused = true;
+        emit pauseSync();
     }
+}
 
-    int screenWidth = trayScreen->geometry().width();
-    int screenHeight = trayScreen->geometry().height();
-    int availableWidth = trayScreen->availableGeometry().width();
-    int availableHeight = trayScreen->availableGeometry().height();
-
-    QPoint topRightDpiAware = QPoint();
-    QPoint topLeftDpiAware = QPoint();
-    if (this->geometry().left() == 0 || this->geometry().top() == 0) {
-        // tray geometry is invalid - QT bug on some linux desktop environments
-        // Use mouse position instead. Cringy, but should work for now
-        topRightDpiAware = QCursor::pos() / trayScreen->devicePixelRatio();
-        topLeftDpiAware = QCursor::pos() / trayScreen->devicePixelRatio();
-    } else {
-        topRightDpiAware = this->geometry().topRight() / trayScreen->devicePixelRatio();
-        topLeftDpiAware = this->geometry().topLeft() / trayScreen->devicePixelRatio();
+/********************************************************************************************/
+/* Helper functions for cross-platform tray icon position and taskbar orientation detection */
+/********************************************************************************************/
+
+QScreen *Systray::currentScreen() const
+{
+    const auto screens = QGuiApplication::screens();
+    const auto cursorPos = QCursor::pos();
+    const auto it = std::find_if(screens.cbegin(), screens.cend(), [=](QScreen *screen) {
+        return screen->geometry().contains(cursorPos);
+    });
+    return (it != screens.cend()) ? *it : nullptr;
+}
+
+int Systray::currentScreenIndex() const
+{
+    const auto screens = QGuiApplication::screens();
+    const auto screenIndex = screens.indexOf(currentScreen());
+    return screenIndex > 0 ? screenIndex : 0;
```

Comment:
Is the final check for the case that `currentScreen()` returns `nullptr`?

In this case I would prefer this code as the function body:
```c++
    const auto screen = currentScreen();
    return screen ? QGuiApplication::screens().indexOf(screen) : 0;
```

Reply:
Now that wouldn't follow the intent of the original author, indexOf() would return -1 if for some weird reason the screens list changed between both calls (of it returned nullptr for some reason). In case of something unexpected to me it looked clear Dominique wanted 0 as the fallback.

Reply:
Just to clarify: 0 as fallback is also what my code does; just the "for some weird reason the screens list changed between both calls" part is not handled, only the `nullptr` case.